vars/detectExecuteScan.groovy

```diff
@@ -0,0 +1,136 @@
+
+@Field Map CONFIG_KEY_COMPATIBILITY = [
+    detect: [
+        projectName: 'projectName',
+        projectVersion: 'projectVersion',
+        scanProperties: 'scanProperties',
+        serverUrl: 'serverUrl',
+        userTokenCredentialsId: 'userTokenCredentialsId'
+    ]
+]
+
+/**
+ * This step executes [Synopsis Detect](https://synopsys.atlassian.net/wiki/spaces/INTDOCS/pages/62423113/Synopsys+Detect) scans.
+ */
+@GenerateDocumentation
+void call(Map parameters = [:]) {
+    handlePipelineStepErrors (stepName: STEP_NAME, stepParameters: parameters) {
+        def script = checkScript(this, parameters) ?: this
+        def utils = parameters.juStabUtils ?: new Utils()
+        // load default & individual configuration
+        Map config = ConfigurationHelper.newInstance(this)
+            .loadStepDefaults()
+            .mixinGeneralConfig(script.commonPipelineEnvironment, GENERAL_CONFIG_KEYS, CONFIG_KEY_COMPATIBILITY)
+            .mixinStepConfig(script.commonPipelineEnvironment, STEP_CONFIG_KEYS,CONFIG_KEY_COMPATIBILITY)
+            .mixinStageConfig(script.commonPipelineEnvironment, parameters.stageName?:env.STAGE_NAME, STEP_CONFIG_KEYS, CONFIG_KEY_COMPATIBILITY)
+            .mixin(parameters, PARAMETER_KEYS, CONFIG_KEY_COMPATIBILITY)
+            .dependingOn('buildTool').mixin('dockerImage')
+            .dependingOn('buildTool').mixin('dockerWorkspace')
+            .withMandatoryProperty('detect/userTokenCredentialsId')
+            .withMandatoryProperty('detect/projectName')
+            .withMandatoryProperty('detect/projectVersion')
+            .use()
+
+        config.stashContent = utils.unstashAll(config.stashContent)
+
+        script.commonPipelineEnvironment.setInfluxStepData('detect', false)
+
+        utils.pushToSWA([
+            step: STEP_NAME,
+            stepParamKey1: 'buildTool',
+            stepParam1: config.buildTool ?: 'default'
+        ], config)
+
+        //prepare Hub Detect execution using package manager
+        switch (config.buildTool) {
+            case 'golang':
+                dockerExecute(script: script, dockerImage: config.dockerImage, dockerWorkspace: config.dockerWorkspace, stashContent: config.stashContent) {
+                    sh 'curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh'
+                    sh 'ln --symbolic $(pwd) $GOPATH/src/hub'
+                    sh 'cd $GOPATH/src/hub && dep ensure'
+                }
+                break
+            default:
+                //no additional tasks are performed
+                echo "[${STEP_NAME}] No preparation steps performed for scan. Please make sure to properly set configuration for `detect.scanProperties`"
+        }
+
+        withCredentials ([string(
+            credentialsId: config.detect.userTokenCredentialsId,
+            variable: 'detectApiToken'
+        )]) {
+            config.detect.scanProperties += [
+                "--blackduck.api.token=${detectApiToken}",
```

Comment:
Token is stored in config and maybe disclosed later on. Add it *on the fly* in line 130.

Reply:
good point, will change that